system/Config/Services.php

```diff
@@ -242,11 +229,7 @@ public static function encrypter($config = null, $getShared = false)
 			return static::getSharedInstance('encrypter', $config);
 		}
 
-		if (empty($config))
-		{
-			$config = new \Config\Encryption();
-		}
-
+		$config     = $config ?? new \Config\Encryption();
```

Comment:
With my argument below of a stricter typehint for configs, is it safe to assume that no BC break will happen if we enforce that also here? 

Or I can change the current typehint of the library to use `Config\Encryption` instead of `BaseConfig`?

Reply:
I think we should go with `Config\Encryption` here too. I really doubt that someone would use something else but we should mention this in the changelog just in case. I mean - we should mention that if we want to provide our own config it should extend the "base class" (`Config\Encryption` in this case). This applies to all services.

Anyway, I would like to hear what others have to say about it. This will probably have to wait till next week because I think I will be alone here this week. So there is no rush :)

Reply:
I think any Service accepting an object that is not a type of that Service's Config should be considered a bug, and no BC concerns. As you mentioned users should be extending Config classes to make their own, so checking the precise Config is desirable and may prevent some cryptic runtime errors with a more helpful error message.